sys/include/net/gnrc/sixlowpan/frag/rb.h

```diff
@@ -172,27 +196,14 @@ static inline void gnrc_sixlowpan_frag_rb_remove(gnrc_sixlowpan_frag_rb_t *rbuf)
     gnrc_sixlowpan_frag_rb_base_rm(&rbuf->super);
     rbuf->pkt = NULL;
 }
-
-/**
- * @brief   Checks if a reassembly buffer entry is complete and dispatches it
- *          to the next layer if that is the case
- *
- * @pre `rbuf != NULL`
- * @pre `netif != NULL`
- *
- * @param[in] rbuf  A reassembly buffer entry. Must not be NULL.
- * @param[in] netif Original @ref gnrc_netif_hdr_t of the last received frame.
- *                  Used to construct the @ref gnrc_netif_hdr_t of the completed
- *                  datagram. Must not be NULL.
- */
-void gnrc_sixlowpan_frag_rb_dispatch_when_complete(gnrc_sixlowpan_frag_rb_t *rbuf,
-                                                   gnrc_netif_hdr_t *netif);
 #else
 /* NOPs to be used with gnrc_sixlowpan_iphc if gnrc_sixlowpan_frag_rb is not
  * compiled in */
-#define gnrc_sixlowpan_frag_rb_remove(rbuf)     (void)(rbuf)
-#define gnrc_sixlowpan_frag_rb_dispatch_when_complete(rbuf, netif) \
-    (void)(rbuf); (void)(netif)
+static inline void gnrc_sixlowpan_frag_rb_remove(gnrc_sixlowpan_frag_rb_t *rbuf)
```

Comment:
The above doc will still be used even if there is no `MODULE_GNRC_SIXLOWPAN_FRAG_RB` defined. In that case, the doc will not really match to what the function is actually doing (nothing).

Reply:
The documentation that generates https://doc.riot-os.org/ ? But that should be independent of what modules are 'used'.

Reply:
@cgundogan if nothing is removed from nothing are you really doing something that is not specified in the doc?

Reply:
(also: this behavior isn't really changed here, but only transformed into an inline function instead of a macro)

Reply:
I would feel comfortable with that if there wasn't a side effect that happens in the function. But as the doc states:
>This functions sets rbuf_t::super::pkt to NULL and removes all rbuf::ints.

That is not happening in this version of the function, is it?

The side effect being to set a struct member of an `input` parameter to `NULL`.

Reply:
>The documentation that generates https://doc.riot-os.org/ ? But that should be independent of what modules are 'used'.

I am fine with the "default" doc, but it is also possible to create a doc tailored to specific build configurations. For this purpose, the doc would be wrong or at least incomplete.

Reply:
Amended doc for the `!MODULE_GNRC_SIXLOWPAN_FRAG_RB` case